### PR TITLE
IGNITE-25031 Remove unnecessary locks from some meta-storage operations

### PR DESCRIPTION
**Pull request by ibessonov**
https://issues.apache.org/jira/browse/IGNITE-25031

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java [hunk 1]
```diff
@@ -565,23 +565,17 @@ public void put(byte[] key, byte[] value, KeyValueUpdateContext context) {
 
     @Override
     public void setIndexAndTerm(long index, long term) {
-        rwLock.writeLock().lock();
-
         try (WriteBatch batch = new WriteBatch()) {
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:568]: Why do we even need a batch here?

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java [hunk 2]
```diff
@@ -1281,8 +1273,6 @@ private Set<UpdateOnlyRevisionEvent> collectUpdateRevisionEventsFromStorage(long
 
     @Override
     public HybridTimestamp timestampByRevision(long revision) {
-        rwLock.readLock().lock();
-
         try {
             assertRequestedRevisionLessThanOrEqualToCurrent(revision, rev);
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1277]: We are accessing a non-volatile field here, are you sure this is safe?

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java [hunk 3]
```diff
@@ -1395,16 +1379,12 @@ protected void saveCompactionRevision(long revision, KeyValueUpdateContext conte
 
     @Override
     public long checksum(long revision) {
-        rwLock.readLock().lock();
-
         try {
             assertRequestedRevisionLessThanOrEqualToCurrent(revision, rev);
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1383]: Same question

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java (hunk 1 above)
Comment (ibessonov) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:568]: I'll check

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java (hunk 2 above)
Comment (ibessonov) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1277]: No, made it volatile

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java (hunk 3 above)
Comment (ibessonov) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1383]: Thank you!

File: modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java [hunk 4]
```diff
@@ -1312,15 +1296,13 @@ public long revisionByTimestamp(HybridTimestamp timestamp) {
 
             byte[] tsValue = rocksIterator.value();
 
-            if (tsValue.length == 0) {
+            if (tsValue.length == 0) { // TODO The fuck? How can it be empty?
```
Comment (sashapolo) [inline modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1299]: Dude...

Comment (ibessonov) [reply modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java:1299]: Sorry

**Review by sashapolo — APPROVED** [review]
